### DISCUSSION TITLE
Route Path Not Following Shapes.txt From Synthetic GTFS Feed

### DISCUSSION THREAD

**Discussion by moss-xyz** [valhalla/valhalla#4367] [Q&A]
Hi all, hoping someone has an idea as to why what I'm trying to do isn't working.

I'm looking to analyze a proposed regional rail network for the city of Atlanta. We created synthetic GTFS files (see [here](https://github.com/valhalla/valhalla/files/13198722/rb1.zip)), which we were able to bring into Valhalla after some troubleshooting: the graph successfully builds, and I can call the **route** and **isochrone** endpoints correctly.

However, weirdly, the output from the **route** endpoint doesn't appear to follow the path of each line defined in **shapes.txt**. Instead, it creates straight-line connections from stop to stop along the route.

This is mainly just an aesthetic thing, but any reason why this might be the case?

My _hunch_ is that it has something to do with the stops themselves not being located directly on the lines? Because this network doesn't exist in real life, I was running into errors in Valhalla no being able to find a suitable edge to snap the stations to for ingress/egress, so I ended up shifting the stations slightly to snap them to the nearest edge in the network myself. Would this decouple it from the shapes file in some way? If so, how would I go about changing it to bring them back together?

**Comment by nilsnolde** [comment]
Yeah, `shapes.txt` is not thoroughly tested. I mean it's tested in code somewhat but likely not real-life tested very much. I'm not sure we perform much/any validation on stops vs shapes locations. Did you try to run [GTFS validation](https://github.com/MobilityData/gtfs-validator) on the data?

Thanks for the test dataset. Any chance you could boil it down to a single route with trips & shape?

Converting to an issue..

**Reply by moss-xyz** [reply]
Sure! Here's just the NGWI route: [ngwi.zip](https://github.com/valhalla/valhalla/files/13199049/ngwi.zip)

**Reply by moss-xyz** [reply]
And yes, I did run it through the validator - it got no critical errors, though a few "stop far from shape" errors that I dismissed for the time being. I think you can see the results [here](https://gtfs-validator-results.mobilitydata.org/41f41469-b651-4345-bd02-ec6dd3ed9df2/report.html)?

**Reply by nilsnolde** [reply]
great thanks! can't promise anything when someone will look at it, but this is very valuable for debugging!

fwiw, best to communicate further on the issue: https://github.com/valhalla/valhalla/issues/4368. (which really should be linked here, dear github)

**Reply by moss-xyz** [reply]
Beaut, let me know if I can help in any way! As I said this is mainly just an aesthetic thing, to generate clean routes, it shouldn't impact the research, so not too bothered by it
